File: CPS-847_Ass2/src/Calculator.java
import java.util.Scanner;

public class Calculator{
	public static void main(String args[]){
		try {
			String oper = "";
			double num1;
			double num2;
			double res = 0;
			if(args.length == 3) {
				num1 = Double.parseDouble(args[0]);
				num2 = Double.parseDouble(args[1]);
				oper = args[2];
			} else {
				Scanner in = new Scanner(System.in);
				System.out.println("Enter your first number: ");
				num1 = in.nextDouble();
				System.out.println("Enter your second number: ");
				num2 = in.nextDouble();
				System.out.println("Enter your operation (add, subtract, multiply, divide)");
				oper = in.next();
				oper = oper.toLowerCase().trim();
			}
			if(oper.equals("add")) {
				res = addThem(num1, num2);
			}
			else if(oper.equals("subtract")) {
				res = subtractThem(num1, num2);
			}
			else if(oper.equals("multiply")) {
				res = multiplyThem(num1, num2);
			}
			else if(oper.equals("divide")) {
				res = divideThem(num1, num2);
			}
			System.out.println("The answer to your operation was: " + res);
			
			
		} catch(Exception e) {
			System.out.println("Something was wrong with your input. Here's the stack trace: ");
			e.printStackTrace();
		}
		
	}
	
	public static double addThem(double a, double b) {
		System.out.println("Adding...");
		return a + b;
	}
	
	public static double subtractThem(double a, double b) {
		System.out.println("Subtracting...");
		return a - b;
	}
	
	public static double multiplyThem(double a, double b) {
		System.out.println("Multiplying...");
		return a * b;
	}
	
	public static double divideThem(double a, double b) {
		System.out.println("Dividing...");
		if(b != 0) return a / b;
		else {
			System.out.println("Can't divide by 0... returning 0");
			return 0;
		}
	}
}

File: CPS847Ass2/CPS847Ass2/Calculator.cs
using System;

namespace CPS847Ass2
{
    public class Calculator
    {
        public static void Main(string[] args)
        {
            try
            {
                string oper = "";
                double num1 = 0;
                double num2 = 0;
                double res = 0;
                if (args.Length == 3)
                {
                    num1 = double.Parse(args[0]);
                    num2 = double.Parse(args[1]);
                    oper = args[2];
                }
                else
                {
                    Console.WriteLine("You need three arguments to run this program");
                }
                if (oper == "add")
                {
                    res = addThem(num1, num2);
                }
                else if (oper == "subtract")
                {
                    res = subtractThem(num1, num2);
                }
                else if (oper == "multiply")
                {
                    res = multiplyThem(num1, num2);
                }
                else if (oper == "divide")
                {
                    res = divideThem(num1, num2);
                }
                Console.WriteLine("The answer to your operation was: " + res);


            }
            catch (Exception e)
            {
                Console.WriteLine("Something was wrong with your input. Here's the stack trace: " + e.ToString());
            }

        }

        public static double addThem(double a, double b)
        {
            Console.WriteLine("Adding...");
            return a + b;
        }

        public static double subtractThem(double a, double b)
        {
            Console.WriteLine("Subtracting...");
            return a - b;
        }

        public static double multiplyThem(double a, double b)
        {
            Console.WriteLine("Multiplying...");
            return a * b;
        }

        public static double divideThem(double a, double b)
        {
            Console.WriteLine("Dividing...");
            if (b != 0) return a / b;
            else
            {
                Console.WriteLine("Can't divide by 0... returning 0");
                return 0;
            }
        }
    }
}
